### PR TITLE
CUSTCOM-133 Code cleanup in security

### PR DESCRIPTION
**Pull request by dmatej**
This PR is based on https://github.com/payara/Payara/pull/4621 and was created to separate testing of both change steps.

# Blockers

* First must be merged the fix https://github.com/payara/Payara/pull/4621 (done)

# Testing

First follow instructions in https://github.com/payara/Payara/pull/4621 , then you can test this in similar way:

This should still pass:
```
git checkout dmatej/CUSTCOM-133-code-cleanup
mvn clean install -PQuickBuild,fastest -TC8
git checkout dmatej/PAYARA-4176-TestContainers-prototype
mvn clean install -Ptest-containers  -pl :test-containers -Dit.test=MetricsSecurityITest,JaspicSamAuthenticationITest # should pass again
```

### CODE REVIEW

**Comment by dmatej** [comment]
Jenkins test please

**Comment by dmatej** [comment]
Jenkins test please

File: nucleus/common/common-util/src/main/java/com/sun/enterprise/security/integration/RealmInitializer.java [hunk 1]
```diff
@@ -37,22 +37,38 @@
...
  */
 public interface RealmInitializer {
 
-    void initializeRealm(Object bundledescriptor, boolean isSystemApp, String realmName);
+    /**
+     * Initializes the internal state of this instance with provided parameters.
+     *
+     * @param bundleDescriptor - contains bundle-specific configuration
+     * @param isSystemApp - realm may have different behavior for system applications
+     * @param defaultRealmName - this realm name will be used as a default if there will not be
```
Comment (Pandrex247) [inline nucleus/common/common-util/src/main/java/com/sun/enterprise/security/integration/RealmInitializer.java:56]:
```suggestion
     * @param defaultRealmName - this realm name will be used as a default if there is not
```
Grammar

File: appserver/web/web-glue/src/main/java/com/sun/enterprise/web/WebModule.java [hunk 2]
```diff
@@ -1266,7 +1264,9 @@ protected void addValve(org.glassfish.web.deployment.runtime.Valve valveDescript
     protected void addCatalinaListener(String listenerName) {
         Object listener = loadInstance(listenerName);
 
-        if ( listener == null ) return;
+        if ( listener == null ) {
```
Comment (Pandrex247) [inline appserver/web/web-glue/src/main/java/com/sun/enterprise/web/WebModule.java:1267]:
```suggestion
        if (listener == null) {
```

File: appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jacc/JaccWebAuthorizationManager.java [hunk 3]
```diff
@@ -181,38 +178,36 @@ public JaccWebAuthorizationManager(WebBundleDescriptor webBundleDescriptor, Serv
...
-        
-        String appname = getAppId();
 
-        postConstruct();
-        
+        String appname = getAppId();
+        SecurityRoleMapperFactoryGen.getSecurityRoleMapperFactory().setAppNameForContext(getAppId(), CONTEXT_ID);
         // Initialize this class and specifically load permissions into the JACC Policy Configuration.
-        initialise(appname);
+        initialize(appname);
```
[the remark this answers is not in the document]
Comment (dmatej) [reply appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jacc/JaccWebAuthorizationManager.java:185]: Funny, I knew that there are differences between uk and us, but I assumed this is a mistake even in UK :D
https://english.stackexchange.com/questions/459553/initialised-or-initialized-which-one-is-correct-spelling
In czech variant of this word is "s" variant obsoleted. I don't want to start any long discussion, but generally, should we prefer us or uk when we are EU-based?

I did some "research" in Payara sources:
image
image
 - probably we should be more consistent at least ...

[the remark this answers is not in the document]
Comment (dmatej) [reply appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jacc/JaccWebAuthorizationManager.java:185]: I will revert it back, as it is not the only requested change and payara sources don't use consistent language, this change can be done in separate issue making the language consistent.

**Comment by dmatej** [comment]
Jenkins test please

**Pull request merged**